client/signup/steps/site-picker/site-picker-submit.jsx

```diff
@@ -0,0 +1,59 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { PLAN_FREE } from 'lib/plans/constants';
+import { getSite } from 'state/sites/selectors';
+import SignupActions from 'lib/signup/actions';
+
+class SitePickerSubmit extends Component {
+	componentWillMount() {
+		const {
+				stepSectionName,
+				stepName,
+				goToStep,
+				selectedSite
+			} = this.props,
+			hasPaidPlan = selectedSite && selectedSite.plan && selectedSite.plan.product_slug !== PLAN_FREE;
```

Comment:
Ah, looking at the internals it looks like it's using data from the plans site specific endpoint. Alright let's hold off on that 👍 . 

In some other PR, we can probably update the selector to use data just off of the site object, which is more likely to be populated.